src/biter/shmutil.h

```diff
@@ -17,18 +17,20 @@ Place, Suite 330, Boston, MA 02111-1307 USA
 #if !defined(SHMUTIL_H_)
 #define SHMUTIL_H_
 
+#include <sys/types.h>
 #include "sheep.h"
 #include "spindle_launch.h"
 
 typedef struct {
    volatile unsigned long *lock;
-   volatile unsigned long *held_by;
+   volatile pid_t *held_by;
    int ref_count;
 } lock_t;
 
 typedef struct {
    volatile int cur_id;
-   unsigned long locks[2];
```

Comment:
I'm nervous about changing the locks[] layout here. I don't remember why I wrote it this way, but I wouldn't have done something this odd without reason. Given that arrays and structs can have different packing (even if on x86_64 it shouldn't pack struct vs array different here), it might have had something to do with that. There's also some memory barriers around the code that uses these arrays, and there's always tough subtleties when those are involved.

That said, I can't re-determine why these should be arrays with just code inspection. And it's too bad I didn't comment why it's like this.

Given that this is a warning cleanup commit, let's not risk subtle semantic changes.

(And this applies to the two other array->struct conversions below as well)

Reply:
Fixed.  Getting the `#include`s to work required updating a couple of `Makefile.am`s.